include/openthread/crypto.h

```diff
@@ -74,6 +74,18 @@ struct otCryptoSha256Hash
  */
 typedef struct otCryptoSha256Hash otCryptoSha256Hash;
 
+#if OPENTHREAD_CONFIG_PSA_CRYPTO_ENABLE
+/**
+ * This function performs HMAC computation.
+ *
+ * @param[in]     aKeyRef        A reference to the key.
+ * @param[in]     aBuf           A pointer to the input buffer.
+ * @param[in]     aBufLength     The length of @p aBuf in bytes.
+ * @param[out]    aHash          A pointer to a `otCryptoSha256Hash` structure to output the hash value.
+ *
+ */
+void otCryptoHmacSha256(uint32_t aKeyRef, const uint8_t *aBuf, uint16_t aBufLength, otCryptoSha256Hash *aHash);
```

Comment:
Regarding public API, we can also use same model of defining a common `struct` type that covers different key types to avoid the need to have different flavors of same API.

Here is my suggestion:
```c
/**
 * This type represents a crypto key.
 *
 */
typedef struct otCryptoKey
{
    const uint8_t *mKey;      ///< Pointer to the buffer containing key. NULL indicates to use `mKeyRef`.
    uint16_t       mKeyLength ///< The key length in bytes (applicable when `mKey` is not NULL).
    uint32_t       mKeyRef;   ///< The PSA key ref (requires `OPENTHREAD_CONFIG_PSA_CRYPTO_ENABLE` and `mKey` to be NULL).
} otCryptoKey;
```

And then have a single API that gets a `otCryptoKey` as input:
```c
void otCryptoHmacSha256(const otCryptoKey *aKey,
                        const uint8_t *     aBuf,
                        uint16_t            aBufLength,
                        otCryptoSha256Hash *aHash);
```

I used a simple struct for `otCryptoKey` and used `mKey` being set to NULL to indicate the "type" of key (whether the key bytes are directly given or if to use the `mKeyRef` value). We can consider using `union` or other data models, but I think such `struct` is simple and easy to use.

Reply:
Agreed.